charts/zalenium/templates/_pod-template.yaml

```diff
@@ -47,7 +47,7 @@ spec:
           protocol: TCP
       livenessProbe:
         httpGet:
-          path: {{ if and (.Values.ingress.path) (not (eq .Values.ingress.path "/")) }}{{ .Values.ingress.path }}{{ end }}/grid/console
+          path: {{ if (.Values.ingress.path) and (not (eq .Values.ingress.path "/")) }}{{ .Values.ingress.path }}{{ end }}/grid/console
```

Comment:
Did you test moving the and in the middle like that ?
Because according to the doc, it does not work like that: https://github.com/helm/helm/blob/master/docs/chart_template_guide/control_structures.md#ifelse

Reply:
Yes I did test it.  When the if condition was like this `path: {{ if and (.Values.ingress.path)  (not (eq .Values.ingress.path "/")) }}{{ .Values.ingress.path }}{{ end }}/grid/console`, I got this error

[rpalaniappan@sdgl15280a331:~/git/zalenium/charts] $ helm upgrade ipteam1 --values=/tmp/values.yaml zalenium
UPGRADE FAILED
ROLLING BACK
Error: render error in "zalenium/templates/deployment.yaml": template: zalenium/templates/deployment.yaml:15:8: executing "zalenium/templates/deployment.yaml" at <include "zalenium.po...>: error calling include: template: zalenium/templates/_pod-template.yaml:50:56: executing "zalenium.podTemplate" at <eq .Values.ingress.p...>: error calling eq: invalid type for comparison
Error: UPGRADE FAILED: render error in "zalenium/templates/deployment.yaml": template: zalenium/templates/deployment.yaml:15:8: executing "zalenium/templates/deployment.yaml" at <include "zalenium.po...>: error calling include: template: zalenium/templates/_pod-template.yaml:50:56: executing "zalenium.podTemplate" at <eq .Values.ingress.p...>: error calling eq: invalid type for comparison```

Reply:
To be honest, even if it seems to work, this syntax is not documented so I prefer to not use it and put us out of the book.
I would instead simplify this expression to `{{ if .Values.ingress.path }}{{ .Values.ingress.path }}{{ end }}/grid/console`. 
I've tested this and it worked for me. 
What do you think ?

Reply:
I'll change to `{{ if .Values.ingress.path }}{{ .Values.ingress.path }}{{ end }}/grid/console`